tool-testsuite/test/org/antlr/v4/test/tool/TestVocabulary.java

```diff
@@ -64,7 +67,8 @@ public void testVocabularyFromTokenNames() {
 		Vocabulary vocabulary = VocabularyImpl.fromTokenNames(tokenNames);
 		Assert.assertNotNull(vocabulary);
 		Assert.assertEquals("EOF", vocabulary.getSymbolicName(Token.EOF));
-		for (int i = 0; i < tokenNames.length; i++) {
+		Assert.assertEquals(vocabulary.getMaxTokenType(), tokenNames.length-1);
+		for (int i = Token.MIN_USER_TOKEN_TYPE; i <= vocabulary.getMaxTokenType(); i++) {
```

Comment:
Yes, the new assertion before the for loop is a good check to add. There is still a problem with the for loop initializer, which now skips checking i==0.